code/datums/traits/negative.dm

```diff
@@ -578,3 +578,52 @@
 	H.remove_language(/datum/language/common)
 	if(!H.can_speak_language(/datum/language/draconic) && !H.can_speak_language(/datum/language/machine))
 		H.grant_language(/datum/language/japanese)
+
+/datum/quirk/allergic
+	name = "Allergic Reaction"
+	desc = "You have had an allergic reaction to medicine in the past. Better stay away from it!"
+	value = -1
+	mob_trait = TRAIT_ALLERGIC
+	gain_text = "<span class='danger'>You remember your allergic reaction to a common medicine.</span>"
+	lose_text = "<span class='notice'>You no longer are allergic to medicine.</span>"
+	medical_record_text = "Patient has a severe allergic reaction to a commong medicine."
+//	var/allergy_list = list(/obj/item/reagent_containers/food/snacks/grown/apple,
+//							/obj/item/reagent_containers/food/snacks/grown/banana
+//							/obj/item/reagent_containers/food/snacks/grown/berries
+//							/obj/item/reagent_containers/food/snacks/grown/cherries
+//							/obj/item/reagent_containers/food/snacks/grown/grapes
+//							/obj/item/reagent_containers/food/snacks/grown/pineapple
+//							/obj/item/reagent_containers/food/snacks/grown/tomato
+							//Was going to make it pick if you allergic to eating a food, or a chem. too much work
+	var/allergy_chem_list = list(	/datum/reagent/medicine/inacusiate,
+									/datum/reagent/medicine/silver_sulfadiazine,
+									/datum/reagent/medicine/styptic_powder,
+									/datum/reagent/medicine/omnizine,
+									/datum/reagent/medicine/oculine,
+									/datum/reagent/medicine/neurine,
+									/datum/reagent/medicine/bicaridine,
+									/datum/reagent/medicine/kelotane,) //Everything in the list can be healed from another source round-start
+	var/reagent_id
+	var/cooldown_time = 600 //1 minute. Cant act again until the first wears off. Test to make sure this is enough
+	var/cooldown = 0
+
+/datum/quirk/allergic/on_spawn()
+	reagent_id = pick(allergy_chem_list)
+	announce_allergy()
+
+/datum/quirk/allergic/proc/announce_allergy()
+	var/datum/reagent/allergy = GLOB.chemical_reagents_list[reagent_id]
+	to_chat(quirk_holder, "<span class='danger'>You remember you are allergic to [allergy.name].</span>")
+
+/datum/quirk/allergic/on_process()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/datum/reagent/allergy = GLOB.chemical_reagents_list[reagent_id]
+	if(cooldown == 0)
+		if(H.reagents.has_reagent(reagent_id))
+			to_chat(quirk_holder, "<span class='danger'>You forgot you were allergic to [allergy.name]!</span>")
+			H.reagents.add_reagent(/datum/reagent/toxin/histamine, 5)
+			cooldown = 1
+			addtimer(CALLBACK(src, .proc/reset_cooldown), cooldown_time)
+
+/datum/quirk/allergic/proc/reset_cooldown()
+	cooldown = 0
```

Comment:
Couples of things here:
1) Combine if statements instead of cascading if possible whenever the condition is short
2) Use TRUE and FALSE instead of 1 and 0 when dealing with boolean values
3) Use VARSET_CALLBACK instead of creating a proc that does nothing but set a variable

Suggestion:
Also im not sure of your intent here, but if you are trying to ensure the original dose wore off before injecting a new dose of histamine, why not just check if there is histamine instead of having a weird cooldown system

```suggestion
	if(H.reagents.has_reagent(reagent_id))
		to_chat(quirk_holder, "<span class='danger'>You forgot you were allergic to [allergy.name]!</span>")
		H.reagents.add_reagent(/datum/reagent/toxin/histamine, 5)
		cooldown = TRUE
		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), cooldown_time)
```

Reply:
I wanted it on a cooldown because there's other sources of histamine.